src/app/api/greenbean/route.ts

```diff
@@ -0,0 +1,75 @@
+import publicClient from "@/lib/publicClient";
+import { NextRequest, NextResponse } from "next/server";
+import ABI from "../../../abis/GREEN_BEAN_ABI.json";
+
+// Contract address for checking green bean status
+const CONTRACT_ADDRESS = "0xdfaa1a2d917df08ea9eae22fec2dd729aa93f97b";
+
+// API Route Handler, return the index for all unclaimed Azukis
+export async function GET(request: NextRequest) {
+  const allBeansClaimed = await getAllGreenBeanNotClaimed();
+  const indexForUnClaimedOGs = getUnclaimedIndices(allBeansClaimed);
+  return NextResponse.json({ indexForUnClaimedOGs });
+}
+
+const cache = new Map<
```

Comment:
great idea of adding the cache! my only concern is ppl snipe OG with green beans on purpose to get the green bean.. we can't entirely get rid of race condition (e.x. ppl compete buying the same one at the same time), but would like to optimize that window.

Reply:
We can add a cache for just 1min or 5min ?

Reply:
yes let's do it for 1 minute, that sounds reasonable to me!